cruise-control-metrics-reporter/src/main/java/com/linkedin/kafka/cruisecontrol/metricsreporter/metric/ContainerMetricUtils.java

```diff
@@ -0,0 +1,122 @@
+            e.printStackTrace();
+        }
+        return proc;
+    }
+
+    private static String readFile(String path) {
+        String s = null;
+        try {
+            s = readStringValue(new FileInputStream(path));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return s;
+    }
+
+    private static String readStringValue(InputStream in) {
+        try {
+            BufferedReader br = new BufferedReader(new InputStreamReader(in, StandardCharsets.UTF_8));
+            String line;
+            while ((line = br.readLine()) != null) {
+                return line;
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    /**
+     * Get the "recent CPU usage" for the JVM process running inside of a container.
+     *
+     * At this time, the methods of OperatingSystemMXBean used for retrieving recent CPU usage are not
+     * container aware and calculate CPU usage with respect to the physical host instead of the operating
+     * environment from which they are called from. There have been efforts to make these methods container
+     * aware but the changes have not been backported to Java versions less than version 14.
+     *
+     * Once these changes get backported, https://bugs.openjdk.java.net/browse/JDK-8226575, we can use
+     * "getSystemCpuLoad()" for retrieving the CPU usage values when running in a container environment.
+     *
+     * @param cpuUtil The "recent CPU usage" for a JVM process with respect to node
+     * @return the "recent CPU usage" for a JVM process with respect to operating environment
+     *         as a double in [0.0,1.0].
+     */
+    public double getContainerProcessCpuLoad(double cpuUtil) {
+        int logicalProcessorsOfNode = getAvailableProcessors();
+        double cpuQuota = getCpuQuota();
+        if (cpuQuota == -1) {
+            /* A CPU quota value of -1 indicates that the cgroup does not adhere to any CPU time restrictions so we
+             * will use the original container agnostic CPU usage value.
+             */
+            return cpuUtil;
+        }
+
+        // Get the number of CPUs of a node that can be used by the operating environment
+        double cpuLimit = (cpuQuota / getCpuPeriod());
+
+        // Get the minimal number of CPUs needed to achieve the reported CPU utilization
+        double cpus = cpuUtil * logicalProcessorsOfNode;
+
+        /* Calculate the CPU utilization of a JVM process with respect to the operating environment.
+         * Since the operating environment will only use the CPU resources allocated by CGroups,
+         * it will always be that: cpuLimit >= cpus and the result is in the [0.0,1.0] interval.
+         */
+        return cpus / cpuLimit;
+    }
+
+    public enum CgroupFiles {
+
+        QUOTA_PATH("/sys/fs/cgroup/cpu/cpu.cfs_quota_us"),
+        PERIOD_PATH("/sys/fs/cgroup/cpu/cpu.cfs_period_us");
```

Comment:
I was getting a `DMI_HARDCODED_ABSOLUTE_FILENAME` spotbugs error when I originally had these pathnames as static variables in the ContainerMetricsUtils class. I just added them back as static variables and referenced them from the CgroupFiles enum to avoid the `DMI_HARDCODED_ABSOLUTE_FILENAME` error. Let me know what you think!